### PR TITLE
Blog post for ZAP baseline action

### PR DESCRIPTION
**Pull request by sshniro**

### CODE REVIEW

File: site/content/blog/2020-04-08-automate-security-testing-with-zap-and-github-actions/index.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
+---
+title: "Automate Security Testing with ZAP and GitHub Actions"
+description: "How ZAP baseline and github actions can help you to automate the security testing"
```
Comment (thc202) [inline site/content/blog/2020-04-08-automate-security-testing-with-zap-and-github-actions/index.md:3]: `GitHub`

File: site/content/blog/2020-04-08-automate-security-testing-with-zap-and-github-actions/index.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+With the increasing number of web application security breaches, it is essential to keep your web application secure at all times. 
+Furthermore having security integrated into your CI/CD pipeline (DevSecOps) will become a lifesaver if you are actively 
+developing the application. To cater to this need ZAP provides a baseline scan feature to find common security faults in 
+a web application without doing any active attacks. 
+
+Furthermore, GitHub Actions makes it easier to automate how to scan and secure web applications at scale. Actions let you 
+write scripts that are triggered based on certain events in your GitHub repo such as — creating a new issue, pushing a 
+commit, or on a scheduled basis. This makes it convenient and easy to automate security testing and to run ZAP scans 
+without involving any of your own infrastructures. The ZAP baseline action is available in the Github Marketplace under
```
Comment (thc202) [inline site/content/blog/2020-04-08-automate-security-testing-with-zap-and-github-actions/index.md:24]: `GitHub`

File: site/content/blog/2020-04-08-automate-security-testing-with-zap-and-github-actions/index.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+Furthermore, GitHub Actions makes it easier to automate how to scan and secure web applications at scale. Actions let you 
+write scripts that are triggered based on certain events in your GitHub repo such as — creating a new issue, pushing a 
+commit, or on a scheduled basis. This makes it convenient and easy to automate security testing and to run ZAP scans 
+without involving any of your own infrastructures. The ZAP baseline action is available in the Github Marketplace under 
+the actions/security category.
+
+The [ZAP baseline-action](https://github.com/marketplace/actions/owasp-zap-baseline-scan) can be configured to periodically 
+scan a publicly available web application. Also, ZAP baseline-action can be configured to public and private repositories as well.
+Based on the scan results ZAP will maintain an active issue in GitHub repository. The action will update the issue if it identifies 
+any new or resolved alerts and will close the issue if all the alerts have been resolved. A detailed report is attached with
```
Comment (thc202) [inline site/content/blog/2020-04-08-automate-security-testing-with-zap-and-github-actions/index.md:30]: `attached to the workflow run`?

File: site/content/blog/2020-04-08-automate-security-testing-with-zap-and-github-actions/index.md [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+write scripts that are triggered based on certain events in your GitHub repo such as — creating a new issue, pushing a 
+commit, or on a scheduled basis. This makes it convenient and easy to automate security testing and to run ZAP scans 
+without involving any of your own infrastructures. The ZAP baseline action is available in the Github Marketplace under 
+the actions/security category.
+
+The [ZAP baseline-action](https://github.com/marketplace/actions/owasp-zap-baseline-scan) can be configured to periodically 
+scan a publicly available web application. Also, ZAP baseline-action can be configured to public and private repositories as well.
+Based on the scan results ZAP will maintain an active issue in GitHub repository. The action will update the issue if it identifies 
+any new or resolved alerts and will close the issue if all the alerts have been resolved. A detailed report is attached with 
+the issue to get more information regarding the identified alerts. The report is available in HTML and markdown formats.
```
Comment (thc202) [inline site/content/blog/2020-04-08-automate-security-testing-with-zap-and-github-actions/index.md:31]: `Markdown`

File: site/content/blog/2020-04-08-automate-security-testing-with-zap-and-github-actions/index.md [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          target: 'https://www.zaproxy.org/'
+```
+
+The ZAP baseline action can be also used to scan a locally built web application. The general idea is to build and serve your web 
+application and make it available via a headless browser so ZAP can access the site via the localhost.
+
+By default, baseline scan reports all alerts as WARNings but you can specify a config file that can change any rules to FAIL or IGNORE.
+The following shows a sample rules file. Create the `rules.tsv` file inside your repository and make sure to update the action
```
Comment (thc202) [inline site/content/blog/2020-04-08-automate-security-testing-with-zap-and-github-actions/index.md:71]: Add example with the field?

File: site/content/blog/2020-04-08-automate-security-testing-with-zap-and-github-actions/index.md [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+authors:
+    - nirojan
+---
+
+With the increasing number of web application security breaches, it is essential to keep your web application secure at all times. 
+Furthermore having security integrated into your CI/CD pipeline (DevSecOps) will become a lifesaver if you are actively 
+developing the application. To cater to this need ZAP provides a baseline scan feature to find common security faults in 
+a web application without doing any active attacks. 
+
+Furthermore, GitHub Actions makes it easier to automate how to scan and secure web applications at scale. Actions let you
```
Comment (thc202) [inline site/content/blog/2020-04-08-automate-security-testing-with-zap-and-github-actions/index.md:21]: Link to https://github.com/features/actions ?

File: site/content/blog/2020-04-08-automate-security-testing-with-zap-and-github-actions/index.md (hunk 5 above)
Comment (sshniro) [reply site/content/blog/2020-04-08-automate-security-testing-with-zap-and-github-actions/index.md]: Do you mean the YAML config to run the action?

File: site/content/blog/2020-04-09-automate-security-testing-with-zap-and-github-actions/index.md [hunk 7]
```diff
@@ -0,0 +1,84 @@
+---
+title: "Automate Security Testing with ZAP and GitHub Actions"
+description: "How ZAP baseline and github actions can help to automate the security testing"
+images:
```
Comment (psiinon) [inline site/content/blog/2020-04-09-automate-security-testing-with-zap-and-github-actions/index.md:4]: How about just using a screenshot of the top part of https://github.com/marketplace/actions/owasp-zap-baseline-scan, ie just he logo, and first part of the 'Action Baseline" section?
I'd put this in the blog near the top as well.

File: site/content/blog/2020-04-09-automate-security-testing-with-zap-and-github-actions/index.md [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+authors:
+    - nirojan
+---
+
+With the increasing number of web application security breaches, it is essential to keep your web application secure at all times. 
+Furthermore having security integrated into your CI/CD pipeline (DevSecOps) will become a lifesaver if you are actively 
+developing the application. To cater to this need ZAP provides a baseline scan feature to find common security faults in 
+a web application without doing any active attacks. 
+
+Furthermore, GitHub Actions makes it easier to automate how to scan and secure web applications at scale. Actions let you
```
Comment (psiinon) [inline site/content/blog/2020-04-09-automate-security-testing-with-zap-and-github-actions/index.md]: 'Furthermore' not needed?

File: site/content/blog/2020-04-08-automate-security-testing-with-zap-and-github-actions/index.md (hunk 5 above)
Comment (thc202) [reply site/content/blog/2020-04-08-automate-security-testing-with-zap-and-github-actions/index.md]: Just the field, e.g.:
```yaml
          rules_file_name: '.github/workflows/rules.tsv'
```

Comment (sshniro) [reply site/content/blog/2020-04-08-automate-security-testing-with-zap-and-github-actions/index.md]: Okay, I have refactored the first yaml to include very minimal configs and in the second part, I have added these fields.

File: site/content/blog/2020-04-09-automate-security-testing-with-zap-and-github-actions/index.md (hunk 7 above)
Comment (sshniro) [reply site/content/blog/2020-04-09-automate-security-testing-with-zap-and-github-actions/index.md:4]: Done

**Comment by sshniro** [comment]
> Worth aloso mentioning that any of the baseline scan options can be set, and linking to https://github.com/zaproxy/zaproxy/wiki/ZAP-Baseline-Scan from somewhere?

Added: Also you can supply any additional baseline [command line options](https://github.com/zaproxy/zaproxy/wiki/ZAP-Baseline-Scan) 
via the `cmd_options` parameter. For example passing `-a` as a command-line argument will include the alpha passive scan rules as well.

File: site/content/blog/2020-04-09-automate-security-testing-with-zap-and-github-actions/index.md (hunk 8 above)
Comment (sshniro) [reply site/content/blog/2020-04-09-automate-security-testing-with-zap-and-github-actions/index.md]: Removed

File: site/content/blog/2020-04-09-automate-security-testing-with-zap-and-github-actions/index.md [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+    steps:
+      - name: ZAP Scan
+        uses: zaproxy/action-baseline@v0.2.0
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          target: 'https://www.zaproxy.org/'
+```
+
+The ZAP baseline action can be also used to scan a locally built web application. The general idea is to build and serve your web 
+application and make it available via a headless browser so ZAP can access the site via the localhost.
```
Comment (thc202) [inline site/content/blog/2020-04-09-automate-security-testing-with-zap-and-github-actions/index.md:67]: Replace `and make it available via a headless browser` (this is not strictly needed) with `locally`?

File: site/content/blog/2020-04-09-automate-security-testing-with-zap-and-github-actions/index.md [hunk 10]
```diff
@@ -0,0 +1,103 @@
...
+          target: 'https://www.zaproxy.org/'
+          rules_file_name: '.zap/rules.tsv'
+          cmd_options: '-a'
+```
+
+Also, you can view the scan logs by navigating to the ZAP scan job.
+
+![scan-job](./images/scan-job.png)
+
+We are excited to here your thoughts and feedback for our new ZAP Baseline Action. Reach us through our dev group or
```
Comment (thc202) [inline site/content/blog/2020-04-09-automate-security-testing-with-zap-and-github-actions/index.md:102]: - `hear`
 - `our` → `the`?

**Pull request merged**